chronos/asyncloop.nim

```diff
@@ -382,17 +383,12 @@ when defined(windows) or defined(nimdoc):
 
   var gDisp{.threadvar.}: PDispatcher ## Global dispatcher
 
-  proc setGlobalDispatcher*(disp: PDispatcher) =
-    ## Set current thread's dispatcher instance to ``disp``.
-    if not gDisp.isNil:
-      doAssert gDisp.callbacks.len == 0
-    gDisp = disp
-
-  proc getGlobalDispatcher*(): PDispatcher =
-    ## Returns current thread's dispatcher instance.
-    if gDisp.isNil:
-      setGlobalDispatcher(newDispatcher())
-    result = gDisp
+  proc setThreadDispatcher*(disp: PDispatcher) {.gcsafe.}
```

Comment:
these forward declarations should move to the non-platform specific part?

Reply:
But `PDispatcher` object is platform specific.